### DISCUSSION TITLE
Is it possible to include `.env` files?

### DISCUSSION THREAD

**Discussion by joaonc** [takluyver/flit#690] [Q&A]
I have an `assets` folder that contains non code stuff, including a `.env` file with some defaults.  In that same folder I also have a `.env.secret` with my personal info that I don't want to be included in the package.

Using the [new style metadata](https://flit.pypa.io/en/stable/pyproject_toml.html?#new-style-metadata), with flit `3.9.0`.

All files in the `assets` folder are being included correctly, except the `.env` (which I want included) and `.env.secret` (which I didn't want included anyway).

Is `flit` excluding these files on purpose?   Is there an option to include the `.env` file?

**Comment by takluyver** [comment]
Are you using `flit build`, or are you using Flit through another frontend, such as `pip` or `python -m build`?

There's an unfortunate discrepancy between those, which I'm planning to clear up but not getting round to. At present, `flit build` defaults to including (in the sdist) everything that's tracked in git, which is what I usually like to upload to PyPI. Using Flit through other frontends defaults to making sdists with only the files needed to make a wheel. Wheels should always contain only the files which will be installed.

However, you can control which files go in the sdist by specifying lists of patterns to include & exclude: https://flit.pypa.io/en/stable/pyproject_toml.html#sdist-section
